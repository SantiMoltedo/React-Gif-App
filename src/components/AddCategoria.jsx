import { useState } from "react"

export const AddCategoria = ({ onNewCategory, actualCategorias }) => {
    const [inputValue, setinputValue] = useState('')

    const onInputChange = (value) => {
        setinputValue(value)
    }

    const onSubmit = (evento) => {
        evento.preventDefault()
        if (inputValue.trim().length < 1 || actualCategorias.includes(inputValue.trim()))
            return
        onNewCategory(inputValue.trim());
        setinputValue("")
    }

    return (
        <form onSubmit={(evento => onSubmit(evento))}>
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={({ target: { value } }) => onInputChange(value)}
            >
            </input>
        </form>
    )
}
