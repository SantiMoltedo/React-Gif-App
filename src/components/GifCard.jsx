import PropTypes from "prop-types"

export const GifCard = ({ gif }) => {
    return (
        <div key={gif.id} className="card">
            <p>{gif.title}</p>
            <img src={gif.url}></img>
        </div>
    )
}

GifCard.propTypes = {
    gif: PropTypes.object.isRequired,
}