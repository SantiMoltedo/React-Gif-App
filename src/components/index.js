export * from './AddCategoria'
export * from './GifCard'
export * from './GifGrid'