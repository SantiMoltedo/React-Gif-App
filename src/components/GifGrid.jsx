import { GifCard } from "./GifCard"
import { useFetchGifs } from "../hooks/useFetchGifs.js"

export const GifGrid = ({ categoria }) => {
    // const [gifs, setgifs] = useState([])

    // const getGifsData = async () => {
    //     const newGifsData = await getGifs(categoria)
    //     setgifs(newGifsData)
    // }

    // useEffect(() => {
    //     getGifsData()
    // }, [])

    const { gifs, isLoading } = useFetchGifs(categoria)

    return (
        <div className="card-grid">
            <h3>{categoria}</h3>
            {isLoading && (<h2>Cargando...</h2>)}
            {gifs.map(gif =>
                <GifCard key={gif.id} gif={gif} />
            )
            }
        </div >
    )
}
