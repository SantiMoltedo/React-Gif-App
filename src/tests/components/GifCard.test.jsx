import { render } from '@testing-library/react'
import { GifCard } from '../../components/GifCard'

describe('Pruebas en componente GifCard', () => {
    const title = 'Naruto'
    const url = 'Naruto'
    const gif = {
        title,
        url,
        id: 0,
    }
    test('Matchear con snapshot', () => {
        const { container } = render(<GifCard gif={gif} />)

        expect(container).toMatchSnapshot();
    })
})