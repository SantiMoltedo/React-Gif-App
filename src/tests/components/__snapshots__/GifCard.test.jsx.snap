// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifCard Matchear con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Naruto
    </p>
    <img
      src="Naruto"
    />
  </div>
</div>
`;
